src/overlays/misc/ovl_kaleido_scope/z_kaleido_scope_PAL.c

```diff
@@ -2470,28 +2512,32 @@ void KaleidoScope_GrayOutTextureRGBA32(u32* texture, u16 pixelCount) {
 void KaleidoScope_UpdateOpening(PlayState* play) {
     PauseContext* pauseCtx = &play->pauseCtx;
 
-    pauseCtx->eye.x += D_8082ABAC[pauseCtx->mode] * ZREG(46);
-    pauseCtx->eye.z += D_8082ABCC[pauseCtx->mode] * ZREG(46);
-    pauseCtx->unk_1EA += 4 * ZREG(46);
+    pauseCtx->eye.x += sPageSwitchEyeDx[pauseCtx->nextPageMode] * ZREG(46);
+    pauseCtx->eye.z += sPageSwitchEyeDz[pauseCtx->nextPageMode] * ZREG(46);
+    pauseCtx->switchPageTimer += 4 * ZREG(46);
 
-    if (pauseCtx->unk_1EA == (64 * ZREG(47))) {
+    if (pauseCtx->switchPageTimer == (4 * 16 * ZREG(47))) {
         // Finished opening
 
         func_80084BF4(play, 1);
-        gSaveContext.buttonStatus[0] = D_8082AB6C[pauseCtx->pageIndex][0];
-        gSaveContext.buttonStatus[1] = D_8082AB6C[pauseCtx->pageIndex][1];
-        gSaveContext.buttonStatus[2] = D_8082AB6C[pauseCtx->pageIndex][2];
-        gSaveContext.buttonStatus[3] = D_8082AB6C[pauseCtx->pageIndex][3];
-        gSaveContext.buttonStatus[4] = D_8082AB6C[pauseCtx->pageIndex][4];
-        pauseCtx->pageIndex = D_8082ABEC[pauseCtx->mode];
+
+        gSaveContext.buttonStatus[0] = gPageSwitchNextButtonStatus[pauseCtx->pageIndex][0];
+        gSaveContext.buttonStatus[1] = gPageSwitchNextButtonStatus[pauseCtx->pageIndex][1];
+        gSaveContext.buttonStatus[2] = gPageSwitchNextButtonStatus[pauseCtx->pageIndex][2];
+        gSaveContext.buttonStatus[3] = gPageSwitchNextButtonStatus[pauseCtx->pageIndex][3];
+        gSaveContext.buttonStatus[4] = gPageSwitchNextButtonStatus[pauseCtx->pageIndex][4];
+
+        pauseCtx->pageIndex = sPageSwitchNextPageIndex[pauseCtx->nextPageMode];
+
         pauseCtx->mainState = PAUSE_MAIN_STATE_IDLE;
         pauseCtx->state++; // PAUSE_STATE_MAIN
+
         pauseCtx->alpha = 255;
         Interface_LoadActionLabelB(play, DO_ACTION_SAVE);
-    } else if (pauseCtx->unk_1EA == 64) {
+    } else if (pauseCtx->switchPageTimer == (4 * 16 * 1)) {
```

Comment:
What do these 3 numbers mean for the timer? The `(4 * 16)` appears a few times

Reply:
The timer increases in increments of 4
`pauseCtx->switchPageTimer += 4 * ZREG(46);`
(`ZREG(46)` is always 1)
-> hence `4*16`

The if before this else has `(4 * 16 * ZREG(47))`
and `ZREG(47)` is always 1
-> so the `4*16*1` is for symmetry

I assume at some point the dev tried `ZREG(47)` == 2 or 3 (or whatever) then this else if would pass at (resp.) 1/2 or 1/3rd of the full animation but idk

(note inside this else if is dead code as the existing "
```
`ZREG(47)` is always 1 so this normally never happens
```
" comment tries to say, is it unclear?)

Reply:
16 is now a macro: `PAGE_SWITCH_NSTEPS`